app/src/main/java/net/frju/flym/data/entities/Filter.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2012-2018 CodingSpiderFox
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+
+package net.frju.flym.data.entities
+
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Index
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+@Entity(tableName = "filters",
+		primaryKeys = ["filterId", "keywordToIgnore"],
+		indices = [(Index(value = ["filterId"]))])
+data class Filter(
+		var filterId: String = "",
```

Comment:
By SharedPres you mean the value set in the settings activity?

Reply:
Yes, the PreferenceFragment will automatically write the value into the SharedPreferences, and you can retrieve it from anywhere. So instead of having a duplicate persistance of this data, I suggest to just use one of them (here the simplest solution is to use the SharedPref and split it into a list of keywords)